### PR TITLE
Update test.support from CPython 3.11.2

### PR DESCRIPTION
**Pull request by youknowone**
blockers
- [ ] #4155
- [x] #4540 

non-blockers
- [ ] #4538
- [ ] #4539 
- [ ] #4541

### CODE REVIEW

File: Lib/test/test_support.py [hunk 1]
```diff
@@ -57,6 +57,8 @@ def test_import_module(self):
         self.assertRaises(unittest.SkipTest,
                           import_helper.import_module, "foo")
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (youknowone) [inline Lib/test/test_support.py:61]: broken due to #4540

File: Lib/test/test_support.py [hunk 2]
```diff
@@ -320,6 +325,8 @@ def test_check_syntax_error(self):
         with self.assertRaises(AssertionError):
             support.check_syntax_error(self, "x=1")
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (youknowone) [inline Lib/test/test_support.py:329]: broken due to #4540

File: Lib/test/test_support.py [hunk 3]
```diff
@@ -691,6 +699,14 @@ def test_print_warning(self):
         self.check_print_warning("a\nb",
                                  'Warning -- a\nWarning -- b\n')
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
+    def test_has_strftime_extensions(self):
+        if support.is_emscripten or sys.platform == "win32":
+            self.assertFalse(support.has_strftime_extensions)
+        else:
+            self.assertTrue(support.has_strftime_extensions)
```
Comment (fanninpm) [inline Lib/test/test_support.py:708]: `support.has_strftime_extensions` will always be `False` when `sys.platform == "win32"`.

I suggest using the os-dependent version:

```python
def test_whatever(self):
    ...

# TODO: RUSTPYTHON
if sys.platform != "win32":
    # TODO: RUSTPYTHON
    test_whatever = unittest.expectedFailure(test_whatever)
```

File: Lib/test/support/__init__.py [hunk 4]
```diff
@@ -474,6 +512,46 @@ def requires_lzma(reason='requires lzma'):
...
+        return unittest.skipUnless(has_socket_support, msg)
+
+# Does strftime() support glibc extension like '%4Y'?
+has_strftime_extensions = False
+if sys.platform != "win32":
+    # bpo-47037: Windows debug builds crash with "Debug Assertion Failed"
+    try:
+        has_strftime_extensions = time.strftime("%4Y") != "%4Y"
+    except ValueError:
+        pass
```
Comment (fanninpm) [inline Lib/test/support/__init__.py:553]: @itsankitkp This is probably what I was looking for in #4474.

File: Lib/test/support/__init__.py [hunk 5]
```diff
@@ -1364,7 +1456,7 @@ def skip_if_buggy_ucrt_strfptime(test):
     global _buggy_ucrt
     if _buggy_ucrt is None:
         if(sys.platform == 'win32' and
-                locale.getdefaultlocale()[1]  == 'cp65001' and
+                locale.getencoding() == 'cp65001' and
```
Comment (fanninpm) [inline Lib/test/support/__init__.py:1459]: This is blocked on #4155.

**Review by fanninpm — APPROVED** [review]
LGTM.

**Pull request merged**